From 0ea2a33d0da17b646f07c44960309b9b9c1dea85 Mon Sep 17 00:00:00 2001
From: Davidlohr Bueso <dbueso@suse.de>
Date: Mon, 29 Feb 2016 14:01:57 -0800
Subject: [PATCH] namespaces: Re-introduce task_nsproxy() helper
Git-commit: 0ea2a33d0da17b646f07c44960309b9b9c1dea85
Patch-mainline: Never, SLES-only API fix
References: bug#963960

With SLE12 commit 7ea3f5e1f9f (namespaces: Use task_lock and
not rcu to protect nsproxy) we removed the helper, which changes
the api. Fix this and reintroduce it, no rcu to be done, so just
return the raw pointer.

Signed-off-by: Davidlohr Bueso <dbueso@suse.de>

---
 include/linux/nsproxy.h | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/include/linux/nsproxy.h b/include/linux/nsproxy.h
index 35fa08f..c8f14a7 100644
--- a/include/linux/nsproxy.h
+++ b/include/linux/nsproxy.h
@@ -62,6 +62,12 @@ extern struct nsproxy init_nsproxy;
  *
  */
 
+/* XXX: SLE12 backward compatibility */
+static inline struct nsproxy *task_nsproxy(struct task_struct *tsk)
+{
+	return tsk->nsproxy;
+}
+
 int copy_namespaces(unsigned long flags, struct task_struct *tsk);
 void exit_task_namespaces(struct task_struct *tsk);
 void switch_task_namespaces(struct task_struct *tsk, struct nsproxy *new);
-- 
2.1.4

